install-scripts/mac/skm-install-mac.sh

```diff
@@ -0,0 +1,128 @@
+#!/bin/bash
+GIT_SKM_REPO=https://github.com/splashkit/skm.git
+MARK_CHECK="\xE2\x9C\x94"
+MARK_CROSS="\xE2\x9C\x95"
+
+HOME_PATH=~
+INSTALL_PATH="${HOME_PATH}/.splashkit"
+
+function has_git() {
+	git --help 2>&1 > /dev/null
+	return $?
+}
+
+function install_git() {
+	echo "macOS developer tools are currently not installed which are required for installation skm."
+	read -p "Would you like to install them now? " -n 1 -r
+	echo ""
+	if [[ $REPLY =~ [Yy]$ ]]
+	then
+		xcode-select --install
+	else
+    	echo "macOS developer tools not installed, please run: \"xcode-select --install\" in the terminal and then reinstall."
+		exit 1
+	fi
+}
+
+echo "Checking for macOS developer tools..."
+has_git
+if [ $? -ne 0 ]; then
+	install_git
+	has_git
+	if [ $? -ne 0 ]; then
+		printf "\t$MARK_CROSS Failed to install macOS developer tools. Please ensure you have \"git\" installed and try again.\n"
+		exit 1
+	fi
+else
+	printf "\t$MARK_CHECK macOS developer tools appears to be installed\n"
+fi
+
+echo "Checking for an existing version of SplashKit..."
+if [ -d "${INSTALL_PATH}" ]; then
+	printf "\t$MARK_CROSS SplashKit is already installed!\n"
+	read -p "Would you like to completely re-install SplashKit? " -n 1 -r
+	echo ""
+	if [[ $REPLY =~ [Yy]$ ]]; then
+		D=`date +%y%m%d-%H%M%S`
+		OLD_PATH="$INSTALL_PATH-$D"
+		echo "Removing existing SplashKit installation..."
+		mv $INSTALL_PATH $OLD_PATH
+		if [ -d "${INSTALL_PATH}" ]; then
+			printf "\t$MARK_CROSS failed to remove existing SplashKit installation.\n"
+			exit 1
+		else
+			printf "\t$MARK_CHECK removed existing SplashKit installation.\n"
+		fi
+	else
+		printf "\t$MARK_CHECK SplashKit is already installed.\n"
+		exit 1
+	fi
+fi
+
+echo "Installing SplashKit..."
+git clone --depth 1 --branch master $GIT_SKM_REPO "${INSTALL_PATH}"
+if [ -d "${INSTALL_PATH}" ]; then
+	printf "\t$MARK_CHECK SplashKit was installed.\n"
+else
+	printf "\t$MARK_CROSS failed to install SplashKit.\n"
+	exit 1
+fi
+
+# Add to .bashrc if using bash and path line is missing for SplashKit inclusion.
+if [ ${SHELL} = "/bin/bash" ]; then
+	echo "Adding \"$INSTALL_PATH\" to your bash profile..."
+	grep -Fqx "export PATH=\"$INSTALL_PATH:\$PATH\"" ~/.bash_profile || echo "export PATH=\"$INSTALL_PATH:\$PATH\"" >> ~/.bash_profile
+    grep -Fqx "export PATH=\"$INSTALL_PATH:\$PATH\"" ~/.bashrc || echo "export PATH=\"$INSTALL_PATH:\$PATH\"" >> ~/.bashrc
```

Comment:
**nitpick:** Issues with indentation? 🤔 It seems that this file is using tab indentation instead of space as the existing files (ie. `skm-install.sh`). Could we stick to space for consistency? Even better, we can add a `.editorconfig` in the root folder to enforce that across IDE. What do you think?